src/ui/inputwin.c

```diff
@@ -927,6 +935,46 @@ _inp_rl_subwin_pagedown_handler(int count, int key)
     return 0;
 }
 
+static int
+_no_op(int count, int key)
+{
+    return 0;
+}
+
+static int (*_inp_rl_scroll_fp)(int count, int key) = _no_op;
+
+void
+inp_update_mouse_prefs(void)
+{
+    if (prefs_get_boolean(PREF_MOUSE))
+        _inp_rl_scroll_fp = _inp_rl_scroll_impl;
+    else
+        _inp_rl_scroll_fp = _no_op;
+}
+
+static int
+_inp_rl_scroll_trampoline(int count, int key)
+{
+    return _inp_rl_scroll_fp(count, key);
+}
+
+static int
+_inp_rl_scroll_impl(int count, int key)
+{
+    if (!prefs_get_boolean(PREF_MOUSE))
+        return 0;
```

Comment:
```suggestion
```
This can go, since this function will only be called if this flag is set to true.